### PR TITLE
Support odo deploy without tag

### PR DESCRIPTION
**Pull request by upLukeWeston**
If the user doesn't provide a tag, rather than bailing out we generate one

### CODE REVIEW

File: pkg/devfile/adapters/kubernetes/component/adapter.go [hunk 1]
```diff
@@ -282,6 +292,27 @@ func (a Adapter) Deploy(parameters common.DeployParameters) (err error) {
 	applicationName := a.ComponentName + "-deploy"
 	deploymentManifest := &unstructured.Unstructured{}
 
+	// oclient.GetImageStream(tag -> a.ComponentName)
+	// TODO: Can we use a occlient created somewhere else rather than create another
```
Comment (EnriqueL8) [inline pkg/devfile/adapters/kubernetes/component/adapter.go]: Create an openshift adapter... :(

File: pkg/devfile/adapters/kubernetes/component/adapter.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ import (
 	"github.com/pkg/errors"
 	"k8s.io/klog"
 
+	imagev1 "github.com/openshift/api/image/v1"
```
Comment (EnriqueL8) [inline pkg/devfile/adapters/kubernetes/component/adapter.go:32]: I don't think we need this

File: pkg/devfile/adapters/kubernetes/component/adapter.go [hunk 3]
```diff
@@ -282,6 +292,27 @@ func (a Adapter) Deploy(parameters common.DeployParameters) (err error) {
 	applicationName := a.ComponentName + "-deploy"
 	deploymentManifest := &unstructured.Unstructured{}
 
+	// oclient.GetImageStream(tag -> a.ComponentName)
+	// TODO: Can we use a occlient created somewhere else rather than create another
+	client, err := occlient.New()
```
Comment (EnriqueL8) [inline pkg/devfile/adapters/kubernetes/component/adapter.go]: Check we are in openshift!

File: pkg/devfile/adapters/kubernetes/component/adapter.go (hunk 2 above)
Comment (upLukeWeston) [reply pkg/devfile/adapters/kubernetes/component/adapter.go:32]: We do need it! For `var imageStream *imagev1.ImageStream`

**Pull request merged**